ipalib/__init__.py

```diff
@@ -942,12 +942,17 @@ def packages(self):
                 ipaserver.plugins,
             )
         else:
-            import ipaclient.remote_plugins
-            import ipaclient.plugins
-            result = (
-                ipaclient.remote_plugins.get_package(self),
-                ipaclient.plugins,
-            )
+            try:
+                import ipaclient.remote_plugins
+                import ipaclient.plugins
+                result = (
+                    ipaclient.remote_plugins.get_package(self),
+                    ipaclient.plugins,
+                )
+            except NetworkError:
+                #  instead of raising the default error connection message,
+                #  raising a more user-friendly one
+                raise GenericError('Cannot find IPA server to contact')
```

Comment:
I don't think that's correct, either. I agree that the ticket's formatting makes it tough to get the grasp of it, but it does not say that the message printed is incorrect, it mentions that it appears at wrong time - when IPA is not installed.

From the BZ:
```
Basically all the commands, if possible should check if ipa
server is installed in the first place and if not then it should display
message like 'IPA is not configured on this system" rather than displaying
Network error or any other message.
```
I believe that we indeed could check whether IPA is actually installed before doing anything, this is already implemented in some of our scripts (e.g. `ipa-ca-install`).

I am not sure, however, what's the scope of such a change, and whether such a change brings enough value given possible time spent to prioritize it.

You should probably also check the discussion at https://github.com/freeipa/freeipa/pull/25 (not sure whether you did already) for some more inputs on this.